materialize-sqlite/sqlite.go

```diff
@@ -0,0 +1,357 @@
+package main
+
+import (
+	"context"
+	stdsql "database/sql"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	pf "github.com/estuary/flow/go/protocols/flow"
+	pm "github.com/estuary/flow/go/protocols/materialize"
+	sql "github.com/estuary/connectors/materialize-sql"
+	boilerplate "github.com/estuary/connectors/materialize-boilerplate"
+	_ "github.com/mattn/go-sqlite3" // Import for register side-effects.
+	log "github.com/sirupsen/logrus"
+)
+
+type config struct {
+	path string
+}
+func (c config) Validate() error {
+	return nil
+}
+
+type tableConfig struct {
+	Table string `json:"table"`
+}
+
+func (c tableConfig) Validate() error {
+	if c.Table == "" {
+		return fmt.Errorf("expected SQLite database configuration `table`")
+	}
+	return nil
+}
+
+func (c tableConfig) GetAdditionalSql() string {
+	return ""
+}
+
+func (c tableConfig) Path() sql.TablePath {
+	return []string{c.Table}
+}
+
+func (c tableConfig) DeltaUpdates() bool {
+	return false // SQLite doesn't support delta updates.
```

Comment:
This probably isn't a big deal now, but I don't see any reason why it _couldn't_ support delta updates.